### ISSUE TITLE
Arity of some rules behaves inconsistently

### ISSUE THREAD

**Issue by codeZeilen** [hpi-swa/Ohm-S#13]
The following rule yields nodes with only four children if the optional part does not match. It yields nodes with five children if it matches. We should check with Ohm/JS if this is intended or accidential behavior.

Pragma = 
  '<' identifier (':' Literal)? '>'

**Comment by codeZeilen** [comment]
A potential fix might be changing PExprOpt>>doEval: to

````Smalltalk
doEval: state

	| originalInputState result input |
	input := state inputStream.
	originalInputState := input getState.
	
	result := self term eval: state.

	result
		ifFalse: [ input setState: originalInputState.
				state 
					addNodeNamed: self class terminalRuleIdentifier
					startedAt: originalInputState 
					withChildren: ((1 to: self arity) collect: [:n | nil]). ].

	^ true
	
````

**Comment by codeZeilen** [comment]
Actually the issue is the following: If the parse succeeds the intermediate results remain on the stack and are not aggregated into one Optional node. If the parse fails it is aggregated into an optional node. Again we have to check Ohm/JS

**Comment by codeZeilen** [comment]
Ohm/JS does yield a node for every node IN an optional node but they might only contain empty strings

**Comment by codeZeilen** [comment]
Fixed in 5030837
